tests/test_device.py

```diff
@@ -724,6 +731,7 @@ def test_outdated_API(self, monkeypatch):
             with pytest.raises(DeviceError, match="plugin requires PennyLane versions"):
                 qml.device("default.qubit", wires=0)
 
+    @pytest.mark.skip(reason="Reloading PennyLane messes with tape mode")
```

Comment:
It didn't seem to fix it, I am completely stumped 🤔 

I have no idea why `importlib.reload(qml)` seems to disable tape mode, as far as I can tell it _shouldn't_.

Reply:
I am not sure but wouldn't it import stuff in `__init__`, then switch tape mode on, and then import stuff in `__init__` again (this time in tape mode) and then switch tape mode on again? That doesn't explain it of course, but should lead to the problems you were trying to avoid by moving tape mode switch to the bottom of `__init__`...

Reply:
The weird thing is, none of those problems I had earlier was replicated here. Instead:

* All tests after this one would be in non-tape mode. Apart from that, everything works fine and all would pass if the test wasn't tape-mode specific.
* In affected tests, `qml.tape_mode_active()` would return False
* In affected tests, `qml.enable_tape()` would have no effect